walkingastro.lic

```diff
@@ -0,0 +1,210 @@
+      end
+      observe
+      return
+    end
+  end
+
+  def observe
+    echo "Observing"
+    case DRC.bput("peer telescope","The pain is too much", "You learned something useful from your observation", "Clouds obscure the sky where the Sun should appear", "Clouds obscure the sky where the Heart should appear", "While the sighting wasn't quite", "You peer aimlessly through your telescope", "You have not pondered your last observation sufficiently","You see nothing regarding the future", "You believe you've learned all that you can about survival", "Too many futures cloud your mind - you learn nothing.")
+    when "You learned something useful from your observation"
+      echo "Successfully observed"
+      waitrt?
+      stow_telescope
+      return
+    when "The pain is too much"
+      echo "You're too wounded"
+      stow_telescope
+      return
+    when "While the sighting wasn't quite"
+      echo "Successfully observed"
+      waitrt?
+      stow_telescope
+      return
+    when "Clouds obscure the sky where the Sun should appear"
+      echo "Piercing Gaze is down"
+      stow_telescope
+      return
+    when "Clouds obscure the sky where the Heart should appear"
+      echo "Piercing Gaze is down"
+      stow_telescope
+      return
+    when "You peer aimlessly through your telescope"
+      echo "Back to determining time"
+      determine_time
+    when "You have not pondered your last observation sufficiently"
+      echo "Too soon, next cycle"
+      waitrt?
+      stow_telescope
+      return
+    when "You see nothing regarding the future"
+      echo "Retry!"
+      waitrt?
+      pause 1
+      observe
+    when "You believe you've learned all that you can about survival"
+      echo "Full pools, head to predict"
+      waitrt?
+      pause 1
+      stow_telescope
+      check_predict
+    when "Too many futures cloud your mind - you learn nothing."
+      echo "Full pools, head to predict"
+      waitrt?
+      pause 1
+      stow_telescope
+      check_predict
+    end
+  end
+
+  def stow_telescope
+    echo "Stowing your telescope"
+    case DRC.bput("stow my telescope","Stow what", "You put your telescope")
+    when "You put your telescope"
+      pause 1
+      return
+    when "Stow what"
+      return
+    end
+  end
+
+  def check_predict
+    echo "Checking whether or not to predict"
+    waitrt?
+    pause 1
+    if DRSkill.getxp('Astrology') > 25
+      echo "Astrology too high, no prediction"
+      return
+    else
+      echo "Time to predict!"
+      fput("align #{@astrology_prediction_skills_survival}")
+      waitrt?
+      pause 1
+      fput("predict future")
+      waitrt?
+      pause 1
+    end
+  end
+
+  def can_observe?
+    return !['buff', 'steal', 'combat-trainer', 'pick', 'craft', 'shape', 'sew','bescort', 'remedy', 'forge', 'carve', 'performance', 'theurgy', 'hlctheurgy', 'astrology', 'astrology2', 'study-art', 'mech-lore', 'first-aid'].any? { |script| Script.running?(script) }
```

Comment:
I'm not a fan of this approach... this list doesn't seem exhaustive nor easy to maintain

Reply:
Can you recommend an alternative? I literally harvested from the ;play script that is being used. Basically, I want walkingastro to not run during any script that can fill your hands with 1 or 2 objects, as a telescope needs both hands to center and peer.

Reply:
I'll have to ponder it. Since this is a new script, people are effectively opting-in to its approach, so this is fine for now.